library/core/functions.general.php

```diff
@@ -1767,6 +1767,23 @@ function ipDecode($packedIP) {
     }
 }
 
+if (!function_exists('ipDecodeWalk')) {
+    /**
+     * Recursively walk through the data, decoding any IP address fields.
+     *
+     * @param array|object $data
+     */
+    function ipDecodeWalk(&$data) {
```

Comment:
ipDecodeRecursive. This is probably a method that can be left on the controller.